File: app/src/main/java/com/blackmirror/dongda/utils/LogUtils.java
package com.blackmirror.dongda.utils;

import android.content.pm.ApplicationInfo;
import android.text.TextUtils;
import android.util.Log;

public class LogUtils {
    public static boolean DEBUG = (AYApplication.getApplication().getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
    public static final String DEFAULT_TAG = "xcx";
    public static final String DEFAULT_ERROR_MSG = "dongda Exception: ";

    public static void d(String message) {
        d(DEFAULT_TAG, message);
    }


    public static void d(Class<?> c, String message) {
        String tag = (c == null ? DEFAULT_TAG : c.getSimpleName());
        d(tag, message);
    }

    public static void d(String TAG, String message) {
        d(TAG, message, null);
    }

    public static void d(String TAG, String message, Throwable t) {
        if (!DEBUG) {
            return;
        }
        if (TextUtils.isEmpty(TAG))
            TAG = DEFAULT_TAG;
        if (t == null) {
            Log.d(TAG, message);
        } else {
            Log.d(TAG, message, t);
        }
    }

    public static void e(Class<?> c, String message){
        String tag = (c == null ? DEFAULT_TAG : c.getSimpleName());
        e(tag, "Exception: "+message, null);
    }

    public static void e(Class<?> c, Throwable t){
        String tag = (c == null ? DEFAULT_TAG : c.getSimpleName());
        e(tag, DEFAULT_ERROR_MSG, t);
    }

    public static void e(Class<?> c, String message, Throwable t) {
        String tag = (c == null ? DEFAULT_TAG : c.getSimpleName());
        e(tag, message, t);
    }

    private static void e(String TAG, String message, Throwable t) {
        if (!DEBUG)
            return;
        if (TextUtils.isEmpty(TAG))
            TAG = DEFAULT_TAG;
        if (t == null) {
            Log.e(TAG, message);
        } else {
            Log.e(TAG, message, t);
        }
    }

}

File: base/src/main/java/com/blackmirror/dongda/utils/LogUtils.kt
package com.blackmirror.dongda.utils

import android.content.pm.ApplicationInfo
import android.util.Log
import com.blackmirror.dongda.base.AYApplication

/**
 * Create By Ruge at 2018-12-07
 */

var DEBUG = AYApplication.me.applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0
const val DEFAULT_TAG = "xcx"
const val DEFAULT_ERROR_MSG = "dongda Exception: "

@JvmOverloads
fun logD(message: String, TAG: String = DEFAULT_TAG, t: Throwable? = null) {
    if (!DEBUG) {
        return
    }
    if (t == null) {
        Log.d(TAG, message)
    } else {
        Log.d(TAG, message, t)
    }
}

fun logE(TAG: String = DEFAULT_TAG, message: String, exception: Throwable? = null) {
    if (!DEBUG)
        return
    if (exception == null) {
        Log.e(TAG, message)
    } else {
        Log.e(TAG, message, exception)
    }
}